src/haystack/encoding/decode_ref_dis_factory.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2020 - 2024, J2 Innovations
+
+use std::borrow::Cow;
+use std::sync::RwLock;
+
+/// Enables display values to be injected into encoded/decoded Ref objects.
+///
+/// This is used to handle the case whereby pre-calculated display
+/// names for Refs need to be utilized when encoding/decoding haystack data.
+///
+/// Please note, the return type is a `Cow<str>`. At some point we'll make `Ref` a little
+/// more flexible regarding how it holds its display data.
+type RefDisFactoryFunc<'a> = Box<dyn Fn(&str, Option<&str>) -> Option<Cow<'a, str>> + Send + Sync>;
+
+/// Holds the Ref factory function used for creating Refs.
+static REF_DIS_FACTORY_FUNC: RwLock<Option<RefDisFactoryFunc>> = RwLock::new(None);
```

Comment:
I suspect that [AtomicPtr](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicPtr.html) would be a better choice here.

Reply:
Thanks for the tip. I think the `RwLock` looks a little cleaner though.

Reply:
Yes, but it has more overhead.